File: Python/BarszczSosnowskiego.py
from Roslina import Roslina
from Organizm import Organizm

class BarszczSosnowskiego(Roslina):
    def __init__(self, x, y, dz, swiat):
        super().__init__(x, y, dz, 10, 0, swiat.IDBARSZCZSOS, swiat)

    def rysowanie(self):
        return "#d45454"

    def nazwa(self):
        return "Barszcz Sosnowskiego"

    def akcja(self):
        if self.GetY() != 0 and self.swiat.GetOrganizmy(self.GetX(), self.GetY() - 1) is not None and self.swiat.GetOrganizmy(self.GetX(), self.GetY() - 1).GetInicjatywa() > 0 and not isinstance(self, type(self.swiat.GetOrganizmy(self.GetX(), self.GetY() - 1))) and self.swiat.GetOrganizmy(self.GetX(), self.GetY() - 1).GetId() != self.swiat.IDCYBEROWCA:
            organizm = self.swiat.GetOrganizmy(self.GetX(), self.GetY() - 1)
            self.swiat.dodajWydarzenie(self, organizm, self.swiat.KILL)
            x = organizm.GetX()
            y = organizm.GetY()
            self.swiat.deletefromList(organizm)
            self.swiat.SetOrganizmy(x, y, None)

        if self.GetY() != self.swiat.Getwyskosc() - 1 and self.swiat.GetOrganizmy(self.GetX(), self.GetY() + 1) is not None and self.swiat.GetOrganizmy(self.GetX(), self.GetY() + 1).GetInicjatywa() > 0 and not isinstance(self, type(self.swiat.GetOrganizmy(self.GetX(), self.GetY() + 1))) and self.swiat.GetOrganizmy(self.GetX(), self.GetY() + 1).GetId() != self.swiat.IDCYBEROWCA:
            organizm = self.swiat.GetOrganizmy(self.GetX(), self.GetY() + 1)
            self.swiat.dodajWydarzenie(self, organizm, self.swiat.KILL)
            x = organizm.GetX()
            y = organizm.GetY()
            self.swiat.deletefromList(organizm)
            self.swiat.SetOrganizmy(x, y, None)

        if self.GetX() != 0 and self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY()) is not None and self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY()).GetInicjatywa() > 0 and not isinstance(self, type(self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY()))) and self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY()).GetId() != self.swiat.IDCYBEROWCA:
            organizm = self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY())
            self.swiat.dodajWydarzenie(self, organizm, self.swiat.KILL)
            x = organizm.GetX()
            y = organizm.GetY()
            self.swiat.deletefromList(organizm)
            self.swiat.SetOrganizmy(x, y, None)

        if self.GetX() != self.swiat.Getszerokosc() - 1 and self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY()) is not None and self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY()).GetInicjatywa() > 0 and not isinstance(self, type(self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY()))) and self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY()).GetId() != self.swiat.IDCYBEROWCA:
            organizm = self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY())
            self.swiat.dodajWydarzenie(self, organizm, self.swiat.KILL)
            x = organizm.GetX()
            y = organizm.GetY()
            self.swiat.deletefromList(organizm)
            self.swiat.SetOrganizmy(x, y, None)

        if self.swiat.Getuklad_planszy() == 6:
            if self.GetX() % 2 == 0:
                if self.GetX() != self.swiat.Getszerokosc() - 1 and self.GetY() != 0 and self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY() - 1) is not None and self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY() - 1).GetInicjatywa() > 0 and not isinstance(self, type(self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY() - 1))):
                    organizm = self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY() - 1)
                    self.swiat.dodajWydarzenie(self, organizm, self.swiat.KILL)
                    x = organizm.GetX()
                    y = organizm.GetY()
                    self.swiat.deletefromList(organizm)
                    self.swiat.SetOrganizmy(x, y, None)

                if self.GetX() != 0 and self.GetY() != 0 and self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY() - 1) is not None and self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY() - 1).GetInicjatywa() > 0 and not isinstance(self, type(self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY() - 1))):
                    organizm = self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY() - 1)
                    self.swiat.dodajWydarzenie(self, organizm, self.swiat.KILL)
                    x = organizm.GetX()
                    y = organizm.GetY()
                    self.swiat.deletefromList(organizm)
                    self.swiat.SetOrganizmy(x, y, None)
            else:
                if self.GetX() != self.swiat.Getszerokosc() - 1 and self.GetY() != self.swiat.Getwyskosc() - 1 and self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY() + 1) is not None and self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY() + 1).GetInicjatywa() > 0 and not isinstance(self, type(self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY() + 1))):
                    organizm = self.swiat.GetOrganizmy(self.GetX() + 1, self.GetY() + 1)
                    self.swiat.dodajWydarzenie(self, organizm, self.swiat.KILL)
                    x = organizm.GetX()
                    y = organizm.GetY()
                    self.swiat.deletefromList(organizm)
                    self.swiat.SetOrganizmy(x, y, None)

                if self.GetX() != 0 and self.GetY() != self.swiat.Getwyskosc() - 1 and self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY() + 1) is not None and self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY() + 1).GetInicjatywa() > 0 and not isinstance(self, type(self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY() + 1))):
                    organizm = self.swiat.GetOrganizmy(self.GetX() - 1, self.GetY() + 1)
                    self.swiat.dodajWydarzenie(self, organizm, self.swiat.KILL)
                    x = organizm.GetX()
                    y = organizm.GetY()
                    self.swiat.deletefromList(organizm)
                    self.swiat.SetOrganizmy(x, y, None)

        super().akcja()

    def stworz_nowe_roslina(self, x, y, sw):
        org = BarszczSosnowskiego(x, y, True, sw)

    def CzyNieZabija(self):
        return False

    def SzansaNaRozsiew(self):
        return 100

File: C++/BarszczSosnowskiego.cpp
#include <iostream>
#include <windows.h>
#include "BarszczSosnowskiego.h"

BarszczSosnowskiego::BarszczSosnowskiego(int x, int y, Swiat* swiat) : Roslina(x, y, 10, 0, IDBARSZCZSOS, swiat)
{
}
BarszczSosnowskiego::BarszczSosnowskiego(int x, int y, bool dz, Swiat* swiat) : Roslina(x, y, dz, 10, 0, IDBARSZCZSOS, swiat)
{
}

void BarszczSosnowskiego::rysowanie()
{
	HANDLE col;
	col = GetStdHandle(STD_OUTPUT_HANDLE);
	SetConsoleTextAttribute(col, KOLORBARSZCZSOS);
	std::cout << "B";
}

std::string BarszczSosnowskiego::nazwa()
{
	return "B";
}

void BarszczSosnowskiego::akcja()
{
	if (this->GetY() != 0 && this->swiat->GetOrganizmy(this->GetX(), this->GetY() - 1) != nullptr && this->swiat->GetOrganizmy(this->GetX(), this->GetY() - 1)->GetInicjatywa() > 0 && typeid(*this) != typeid(*this->swiat->GetOrganizmy(this->GetX(), this->GetY() - 1)))
	{
		Organizm* organizm = this->swiat->GetOrganizmy(this->GetX(), this->GetY() - 1);
		this->swiat->dodajWydarzenie(this, organizm, KILL);
		int x = organizm->GetX();
		int y = organizm->GetY();
		this->swiat->deletefromList(organizm);
		this->swiat->SetOrganizmy(x, y, nullptr);
	}
	if (this->GetY() != this->swiat->Getwyskosc() - 1 && this->swiat->GetOrganizmy(this->GetX(), this->GetY() + 1) != nullptr && this->swiat->GetOrganizmy(this->GetX(), this->GetY() + 1)->GetInicjatywa() > 0 && typeid(*this) != typeid(*this->swiat->GetOrganizmy(this->GetX(), this->GetY() + 1)))
	{
		Organizm* organizm = this->swiat->GetOrganizmy(this->GetX(), this->GetY() + 1);
		this->swiat->dodajWydarzenie(this, organizm, KILL);
		int x = organizm->GetX();
		int y = organizm->GetY();
		this->swiat->deletefromList(organizm);
		this->swiat->SetOrganizmy(x, y, nullptr);
	}
	if (this->GetX() != 0 && this->swiat->GetOrganizmy(this->GetX() - 1, this->GetY()) != nullptr && this->swiat->GetOrganizmy(this->GetX() - 1, this->GetY())->GetInicjatywa() > 0 && typeid(*this) != typeid(*this->swiat->GetOrganizmy(this->GetX() - 1, this->GetY())))
	{
		Organizm* organizm = this->swiat->GetOrganizmy(this->GetX() - 1, this->GetY());
		this->swiat->dodajWydarzenie(this, organizm, KILL);
		int x = organizm->GetX();
		int y = organizm->GetY();
		this->swiat->deletefromList(organizm);
		this->swiat->SetOrganizmy(x, y, nullptr);
	}
	if (this->GetX() != this->swiat->Getszerokosc() - 1 && this->swiat->GetOrganizmy(this->GetX()  + 1, this->GetY()) != nullptr && this->swiat->GetOrganizmy(this->GetX() + 1, this->GetY())->GetInicjatywa() > 0 && typeid(*this) != typeid(*this->swiat->GetOrganizmy(this->GetX() + 1, this->GetY())))
	{
		Organizm* organizm = this->swiat->GetOrganizmy(this->GetX() + 1, this->GetY());
		this->swiat->dodajWydarzenie(this, organizm, KILL);
		int x = organizm->GetX();
		int y = organizm->GetY();
		this->swiat->deletefromList(organizm);
		this->swiat->SetOrganizmy(x, y, nullptr);
	}

	Roslina::akcja();
}

void BarszczSosnowskiego::stworz_nowe_Roslina(int x, int y, Swiat* sw)
{
	Organizm* org = new BarszczSosnowskiego(x, y, true, sw);
}

bool BarszczSosnowskiego::CzyNieZabija()
{
	return false;
}


int BarszczSosnowskiego::SzansaNaRozsiew()
{
	return 100;
}

BarszczSosnowskiego::~BarszczSosnowskiego()
{
}